linux/datadog_wrapper

```diff
@@ -67,30 +84,78 @@ setUpNodeEnv() {
 setUpDotnetEnv() {
     echo "Setting up Datadog tracing for .NET"
 
-    setUpCommonEnv || return
+    getBinaries || return
 
     if [ -z "$DD_DOTNET_TRACER_VERSION" ]; then
         DD_DOTNET_TRACER_VERSION=2.19.0
     fi
+    
+    echo "Using version $DD_DOTNET_TRACER of the .NET tracer"
     DD_DOTNET_TRACER_FILE=datadog-dotnet-apm-${DD_DOTNET_TRACER_VERSION}.tar.gz
     DD_DOTNET_TRACER_URL=https://github.com/DataDog/dd-trace-dotnet/releases/download/v${DD_DOTNET_TRACER_VERSION}/${DD_DOTNET_TRACER_FILE}
 
     echo "Installing .NET tracer from $DD_DOTNET_TRACER_URL"
-    curl -L $DD_DOTNET_TRACER_URL -o $DD_DOTNET_TRACER_FILE|| return
-    tar -xzf $DD_DOTNET_TRACER_FILE || return
+    curl_or_log_failure $DD_DOTNET_TRACER_URL $DD_DOTNET_TRACER_FILE
 
     export CORECLR_ENABLE_PROFILING=1
     export CORECLR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
     export CORECLR_PROFILER_PATH=${DD_DIR}/Datadog.Trace.ClrProfiler.Native.so
     export DD_DOTNET_TRACER_HOME=${DD_DIR}
 }
 
-if [ "$WEBSITE_STACK" = "NODE" ]; then
-    setUpNodeEnv;
-elif [ "$WEBSITE_STACK" = "DOTNETCORE" ]; then
-    setUpDotnetEnv;
-fi
+setUpTraceAgent() {
+    REMOTE_TRACE_AGENT="trace-agent-${DD_AAS_LINUX_VERSION}.tar.gz"
+    LOCAL_TRACE_AGENT="trace-agent"
+
+    # Curl for the binaries or skip if the most recent version is already present
+    check_file_exists_or_curl $REMOTE_TRACE_AGENT $LOCAL_TRACE_AGENT
+
+    echo "Starting trace agent"
+    ./trace-agent &
+    TRACER_PID=$!
+    trap 'kill $TRACER_PID ; exit' INT
+}
+
+setUpDogStatsD() {
+    REMOTE_DOGSTATSD="dogstatsd-${DD_AAS_LINUX_VERSION}.tar.gz"
+    LOCAL_DOGSTATSD="dogstatsd"
+
+    check_file_exists_or_curl $REMOTE_DOGSTATSD $LOCAL_DOGSTATSD
+    
+    echo "Starting Dogstatsd agent"
+    ./dogstatsd start &
+    DSD_PID=$!
+    trap 'kill $DSD_PID ; exit' INT
+}
+
+check_file_exists_or_curl() {
+    # We don't want to curl the binary if the same version is already present in AAS
+    # $1 is the remote filename on github, eg trace-agent-${DD_AAS_LINUX_VERSION}.tar.gz
+    # $2 is the local filename after the script has completed
+    if [ ! -f "$1" ]; then 
```

Comment:
It would be helpful to save the cmdline arguments as named variables. Would make it easier for me to understand what you expect their values to be. Something like:

```bash
cookies=$1
apples=$2
```